Added optimization environment variables and created a non-pip reliant TF install

dphow opened the pull request:
- Updated README.md
- Added a non-pip reliant version and separated install options into tf_pip and tf_nopip folders. 
- Added two optimization environment variables in `activate_env_vars.sh` and `deactivate_env_vars.sh`  as described below by Rory:
*****
There are a number of run-time settings that I've found beneficial for TF GPU performance on Casper and Gust. Most importantly:
TF_GPU_THREAD_MODE=gpu_private
and
TF_GPU_THREAD_COUNT=2 (default) or sometimes 4 seems a little better.

For some random run-time settings you can try for TF performance, look at some of the options here
[https://github.com/NVIDIA/DeepLearningExamples/issues/57](https://github.com/NVIDIA/DeepLearningExamples/issues/57)
*****

Please advise if you prefer I separate these install approaches and changes. Personally, I find pip to be difficult to maintain, even though Google officially recommends that all TF installs utilize pip. The `conda-forge` versions of TF are community maintained but I find those versions to be pretty reliable, though often a version or two behind Google's main releases via pip.

djgagne commented on tf_pip/tf212_env.yml:
```diff
@@ -4,11 +4,11 @@ channels:
-  - numpy=1.23.*
-  - cudatoolkit=11.8.0
+  - libblas=*=*mkl
+  - numpy
```
Will numpy install properly if you remove the 1.23.* designation? The pip installation of tensorflow downgrades numpy but then does not downgrade scipy or pandas, resulting in errors.

djgagne reviewed (commented):
Thank you for adding the nopip option to the installation process and updating the documentation. I suggest keeping the numpy 1.23.* requirement in the nopip version of the directory for now because leaving it out causes pip to downgrade numpy when installing tensorflow without also downgrading scipy and pandas, causing import errors.

dphow commented:
I changed the numpy version to install by conda before the pip install of TF to match what pip requires. This appears to resolve the issue of matching numpy versions to scipy and pandas.

I also updated the install options to include more recent TF release versions for each process, updating the `README.md` as appropriate.
